src/Symfony/Component/HttpClient/CurlHttpClient.php

```diff
@@ -0,0 +1,314 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Component\HttpClient\Exception\InvalidArgumentException;
+use Symfony\Component\HttpClient\Exception\TransportException;
+use Symfony\Component\HttpClient\Response\CurlResponse;
+use Symfony\Component\HttpClient\Response\ResponseStream;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\HttpClientTrait;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Symfony\Contracts\HttpClient\ResponseStreamInterface;
+
+/**
+ * A performant implementation of the HttpClientInterface contracts based on the curl extension.
+ *
+ * This provides fully concurrent HTTP requests, with transparent
+ * HTTP/2 push when a curl version that supports it is installed.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class CurlHttpClient implements HttpClientInterface
+{
+    use HttpClientTrait;
+
+    private $defaultOptions = self::OPTIONS_DEFAULTS;
+    private $noProxy;
+    private $multi;
+
+    /**
+     * @param array $defaultOptions     Default requests' options
+     * @param int   $maxHostConnections The maximum number of connections to a single host
+     * @param int   $noProxy            A comma separated list of hosts that do not require a proxy to be reached, even if one is specified
+     */
+    public function __construct(array $defaultOptions = [], int $maxHostConnections = 6, string $noProxy = null)
```

Comment:
`?string $noProxy = null` ?

Reply:
@nicolas-grekas Actually it looks like you do this on purpose so I won't comment on every occurence.

Reply:
That's not consistent with the code style of the code base. We don't repeat the info, null default values have exactly the same meaning.